packages/compiler/src/core/program.ts

```diff
@@ -260,6 +264,67 @@ export async function compile(
 
   return program;
 
+  function isProjectionUsed() {
+    const isProjectionStatement = (node: Node): true | undefined => {
+      if (node.kind === SyntaxKind.ProjectionStatement) {
+        return true;
+      }
+      return visitChildren(node, isProjectionStatement);
+    };
+    for (const file of program.sourceFiles.values()) {
+      if (program.getSourceFileLocationContext(file.file).type === "project") {
+        if (visitChildren(file, isProjectionStatement) === true) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  function validateUnnecessaryCode() {
+    // Don't provide unused diagnostics if customer is using projection in the project because
+    // the projection statements will only be processed when applying projection. There is no way to determine
+    // whether "import" or "using" is referenced from them, so we just skip here to avoid providing incorrect suggestions (diagnostics)
+    // This should be fine for now considering projection is an experiemental feature.
```

Comment:
@timotheeguerin , I tried to check the project part code and can't find a way to identify the references as commented here. Please let me know if there is some way I missed to count them considering my knowledge in compiler is still limited. thanks.